packages/react-query/src/internals/context.tsx

```diff
@@ -240,5 +241,27 @@ export interface TRPCQueryUtils<TRouter extends AnyRouter> {
   getInfiniteQueryData: (
     queryKey: TRPCQueryKey,
   ) => InfiniteData<unknown> | undefined;
+
+  /**
+   * @link https://tanstack.com/query/latest/docs/reference/QueryClient/#queryclientsetmutationdefaults
+   */
+  setMutationDefaults: (
+    mutationKey: [string[]],
```

Comment:
iirc our mutationKeys use the same structure?

```suggestion
    mutationKey: TRPCQueryKey,
```

Reply:
> iirc our mutationKeys use the same structure?

Yeah, I wasn’t sure about that. I wrote something here: https://github.com/trpc/trpc/pull/5814#discussion_r1648223193

`TRPCQuery` is fine because it’s defined as
```ts
export type TRPCQueryKey = [
  string[],
  { input?: unknown; type?: Exclude<QueryType, 'any'> }?,
]
```
But if I recall correctly, the second element of the tuple (the object) isn’t meaningful for mutations and isn’t used at all. You can check this in the `useMutation` definition (`createHooksInternal.tsx`) where the mutationKey is just `[path]`.

So it’s just a matter of semantics/preferences. If you prefer `TRPCQuery`, it’s perfectly fine.

Reply:
At the very least I think it should be a named type `TRPCMutationKey` placed next to `TRPCQueryKey` with the difference in semantics laid out

Reply:
Ok perfect! We can define it as:
```ts
export type TRPCMutationKey = [
  TRPCQueryKey[0]
];
```
to highlight how the two types are related, or simply:
```ts
export type TRPCMutationKey = [
  string[]
];
```

Do you think we should also update the return type of the `getQueryKey` function? We could use `TRPCMutationKey` when the input extends `DecoratedMutation<any>` and `TRPCQueryKey` otherwise. However, in that case, the name of the function itself, `getQUERYKey`, might be confusing

Reply:
I think 2 functions would be best

Reply:
Okay, everything's clear. I’ll proceed then

Reply:
Which option do you prefer:

- Export `getMutationKey` and `TRPCMutationKey` from `@trpc/react-query/getQueryKey`
- Export `getMutationKey` from `@trpc/react-query/getMutation` (and add a new documentation page)
- Deprecate `@trpc/react-query/getQueryKey` and export `getQueryKey`, `getMutationKey`, `TRPCQueryKey`, and `TRPCMutationKey` from a new module `@trpc/react-query/getKey` (or a better name if available)

Reply:
The former sounds good to me but no strong opinion, and also re-export it from the main entrypoint for public consumption

Reply:
Added everything in https://github.com/trpc/trpc/pull/5814/commits/5322e8875b0d36dba466b0a99f6500c01fe1a34a